includes/class-wc-calypso-bridge-admin-setup-checklist.php

```diff
@@ -101,13 +116,294 @@ public function admin_menu_structure() {
+				'completed_title' => __( 'View Settings', 'wc-calypso-bridge' ),
+				'description' => __( 'Integrating Facebook with your store and drive sales.', 'wc-calypso-bridge' ),
+				'estimate' => '20',
+				'link' => 'admin.php?page=wc-settings&tab=integration&section=facebookcommerce',
+				'learn_more' => 'https://www.facebook.com/business/help/900699293402826',
+				'condition' => false, // TODO Condition logic here.
+				'extension' => 'facebook-for-woocommerce/facebook-for-woocommerce.php',
+			),
+
+			array(
+				'title' => __( 'Integrate with Mailchimp', 'wc-calypso-bridge' ),
+				'completed_title' => __( 'View Settings', 'wc-calypso-bridge' ),
+				'description' => __( 'Connect your store to bring the power of email marketing to your business.', 'wc-calypso-bridge' ),
+				'estimate' => '20',
+				'link' => 'options-general.php?page=mailchimp-woocommerce',
+				'learn_more' => 'https://wordpress.org/plugins/mailchimp-for-woocommerce/',
+				'condition' => false, // TODO Condition logic here.
+				'extension' => 'mailchimp-for-woocommerce/mailchimp-woocommerce.php',
+			),
+		);
+
+		$completed = 0;
+		$tasks     = array();
+		foreach ( $all_tasks as $task ) {
+			// Remove tasks for extensions that are not active.
+			if ( isset( $task['extension'] ) ) {
+				if ( ! is_plugin_active( $task['extension'] ) ) {
+					continue;
+				}
+			}
+
+			$tasks[] = $task;
+			if ( true === $task['condition'] ) {
+				$completed++;
+			}
+		}
+		$total = count( $tasks );
+
+		return( array(
+			'tasks'       => $tasks,
+			'completed'   => $completed,
+			'uncompleted' => $total - $completed,
+			'total'       => $total,
+		) );
 	}
 
+	/**
+	 * Renders the checklist display.
+	 */
+	public function checklist() {
+		$data = $this->get_task_data();
+		$percentage = floor( ( $data['completed'] / $data['total'] ) * 100 );
+		?>
+			<div class="checklist">
+				<div class="checklist-card checklist__header is-compact">
+					<div class="checklist__header-main">
+						<div class="checklist__header-progress">
+							<h2 class="checklist__header-progress-text"><?php esc_html_e( 'Your setup list', 'wc-calypso-bridge' ); ?></h2>
+							<span class="checklist__header-progress-number"><?php echo esc_html( $data['completed'] ); ?>/<?php echo esc_html( $data['total'] ); ?></span>
+						</div>
+						<div class="progress-bar is-compact">
+							<div class="progress-bar__progress" style="width: <?php echo intval( $percentage ); ?>%;"></div>
+						</div>
+					</div>
+				</div>
+				<div class="checklist__tasks">
+					<?php
+					foreach ( $data['tasks'] as $task ) {
+						$this->render_task( $task );
+					}
+					?>
+				</div>
+			</div>
+		<?php
+	}
+
+	/**
+	 * Renders a specific task.
+	 *
+	 * @param array $task Array of task information.
+	 */
+	public function render_task( $task ) {
+		$task_url = admin_url( $task['link'] );
+		?>
+		<div class="checklist-card checklist__task has-actionlink is-compact
+		<?php
+		if ( true === $task['condition'] ) {
```

Comment:
Can probably shorthand this to make a bit more readable `<?php echo $task['condition'] ? 'is-complete: ''; ?>`